src/persist-client/src/batch.rs

```diff
@@ -223,16 +224,21 @@ pub struct BatchBuilderConfig {
     pub(crate) stats_untrimmable_columns: Arc<UntrimmableColumns>,
 }
 
+// TODO: Remove this once we're comfortable that there aren't any bugs.
+pub(crate) const BATCH_DELETE_ENABLED: Config<bool> = Config::new(
+    "persist_batch_delete_enabled",
+    false,
+    "Whether to actually delete blobs when batch delete is called (Materialize).",
+);
```

Comment:
Style thought: the `new` constructor feels a little Go-y—looks a lot like the Go flags package. I feel like it'd be slightly more Rust-y to use a params struct:

```rust
Config::new(ConfigSpec {
    name: "persist_batch_delete_enabled",
    default: false,
    description: "Whether to actually delete blobs when batch delete is called (Materialize).",
});
```

It's slightly more work to type out and slightly more verbose, but I find it slightly faster to understand what each of the parameters is while reading the code.

Reply:
Weird! Seems like a `rustc` bug, or at least a limitation.

> Another option is a `config_{int,..}!` macro, which might anyway be something we need to add if we do the global static registry. Ditto if we want the file and line number of the config definition to be available.

Makes sense. Happy to defer this conversation to a later PR that adds the global registry.

Reply:
> Weird! Seems like a rustc bug, or at least a limitation.

Yeah, I got the same vibes.

> Happy to defer this conversation to a later PR that adds the global registry.

I might take you up on this. Another option (disclaimer: that I haven't tried, no idea if this works) is a sketchy `std::mem::forget` and a comment in `new`.